docs/developers.md

```diff
@@ -20,80 +38,297 @@ For completed APB examples, take a look at some of the APBs in the [ansibleplayb
+│       ├── defaults
+│       │   └── main.yml
+│       └── tasks
+│           └── main.yml
+└── playbooks/
+    └── provision.yml
+    └── deprovision.yml
+    └── bind.yml
+    └── unbind.yml
 ```
-### Spec File (Version 1.0)
-The `apb init` will auto generate a spec file.  You must edit this spec file to fit your application. The following is the spec file for [etherpad-apb](https://github.com/ansibleplaybookbundle/etherpad-apb).  More examples can be found in the [ansibleplaybookbundle org](https://github.com/ansibleplaybookbundle)
 
-```yml
+# APB Spec File
+The APB Spec File (`apb.yml`) is where the outline of your application is declared.  The following is an example APB spec
+
+```yaml
 version: 1.0
-name: etherpad-apb
-description: Note taking web application
+name: example-apb
+description: A short description of what this APB does
 bindable: True
 async: optional
 metadata: 
-  documentationUrl: https://github.com/ether/etherpad-lite/wiki
-  imageUrl: https://translatewiki.net/images/thumb/6/6f/Etherpad_lite.svg/200px-Etherpad_lite.svg.png
-  dependencies: ['docker.io/mariadb:latest', 'docker.io/tvelocity/etherpad-lite:latest']
-  displayName: Etherpad (APB)
-  longDescription: An apb that deploys Etherpad Lite
+  documentationUrl: <link to documentation>
+  imageUrl: <link to URL of image>
+  dependencies: ['<registry>/<organization>/<dependency-name-1>', '<registry>/<organization>/<dependency-name-2>']
+  displayName: Example App (APB)
+  longDescription: A longer description of what this APB does
   providerDisplayName: "Red Hat, Inc."
 plans:
   - name: default
-    description: A single etherpad application with no DB
+    description: A short description of what this plan does
     free: true
     metadata:
       displayName: Default
-      longDescription: This plan provides a single Etherpad application with no database
+      longDescription: A longer description of what this plan deploys
       cost: $0.00
     parameters:
-      - name: mariadb_name
-        required: true
-        default: etherpad
-        type: string
-        title: MariaDB Database Name
-      - name: mariadb_user
+      - name: parameter_one
         required: true
-        default: etherpad
-        title: MariaDB User
+        default: foo_string
         type: string
+        title: Parameter One
         maxlength: 63
-      - name: mariadb_password
-        default: admin
-        type: string
-        description: A random alphanumeric string if left blank
-        title: MariaDB Password
-      - name: mariadb_root_password
-        default: admin
-        type: string
-        description: root password for mariadb
-        title: Root Password
+      - name: parameter_two
+        required: true
+        default: true
+        title: Parameter Two
+        type: boolean
 ```
+## Top level structure
+* `version`: Version of the APB spec. Please see [versioning](#apb-spec-versioning) for more information.
+* `name`: Name of the APB.
```

Comment:
are there restrictions? if so let's add that here. I don't remember if there were any

Reply:
Yes there are. [Here[(https://docs.docker.com/engine/reference/commandline/tag/#usage) is Docker's restriction on the tag name. I will update.